src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		if (this.version >= 3) {
+			validatePayload(this.payload.length >= 4);
+		} else {
+			validatePayload(this.payload.length >= 2);
+		}
+		this._colorComponent = this.payload[0];
+		this._currentValue = this.payload[1];
+
+		if (this.version >= 3) {
+			this._targetValue = this.payload[2];
+			this._duration = Duration.parseReport(this.payload[3]);
+		}
+
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const colorName = getColorKeyFromComponent(this._colorComponent);
+		if (!colorName) {
+			return;
+		}
+
+		const valueId: ValueID = {
+			commandClass: CommandClasses["Color Switch"],
+			property: "currentColor",
+			endpoint: endpoint.index,
+			propertyKey: colorName,
+		};
+
+		// TODO: Should we be setting this every run?
+		node.valueDB.setMetadata(valueId, {
+			...ValueMetadata.ReadOnlyNumber,
+			label: `Color ${colorName}`,
+			description: `The current level of the ${colorName} color.`,
+			min: 0,
+			max: 255,
+		});
+		node.valueDB.setValue(valueId, this._currentValue);
+
+		if (this._targetValue != undefined) {
+			const targetValueId: ValueID = {
+				commandClass: CommandClasses["Color Switch"],
+				property: "targetColor",
+				endpoint: endpoint.index,
+				propertyKey: colorName,
+			};
+			node.valueDB.setMetadata(targetValueId, {
+				...ValueMetadata.Number,
+				label: `Color ${colorName}`,
+				description: `The target level of the ${colorName} color.`,
+				min: 0,
+				max: 255,
+			});
+			node.valueDB.setValue(targetValueId, this._targetValue);
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+
+	private _currentValue: number;
+	public get currentValue(): number {
+		return this._currentValue;
+	}
+
+	private _targetValue: number | undefined;
+	public get targetValue(): number | undefined {
+		return this._targetValue;
+	}
+
+	private _duration: Duration | undefined;
+	public get duration(): Duration | undefined {
+		return this._duration;
+	}
+}
+
+interface ColorSwitchCCGetOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+}
+
+const testResponseForColorSwitchGet: CCResponsePredicate = (
+	sent: ColorSwitchCCGet,
+	received,
+	isPositiveTransmitReport,
+) => {
+	return received instanceof ColorSwitchCCReport &&
+		sent.colorComponent === received.colorComponent
+		? "final"
+		: // TODO: What is isPositiveTransmitReport?  Do I need it?
```

Comment:
Yep, you need it - it tells us that the message was received. Just ignore the part after `? "final"` - I'm not happy with this check design, but it is working ;)